sys_conn_df_darwin.go

```diff
@@ -47,9 +47,8 @@ func setDF(rawConn syscall.RawConn) (bool, error) {
 	return true, nil
 }
 
-func isMsgSizeErr(err error) bool {
-	return errors.Is(err, unix.EMSGSIZE)
-}
+func isSendMsgSizeErr(error) bool { return false }
```

Comment:
should this be `unix.EMSGSIZE`?

Reply:
If I understand EMSGSIZE correctly, it only applies on the receive side: That's why I split `isMsgSizeErr` into a send and a receive part.

On Linux, when a large packet is received, the kernel will just return you whatever fits into the buffer. This is fine, since QUIC packets are authenticated, and too short packets will be discarded.

Reply:
Respectfully, I think you might have that backwards- `sendmsg()` will return `EMSGSIZE` on both Darwin and Linux when the message exceeds what it thinks the MTU is. `recvmsg()` can also return `EMSGSIZE`, but it only does so for `iov_msglen <= 0` or `iov_msglen > IOV_MAX`. `MSG_TRUNC` only applies on the receive side though.

Reply:
https://man7.org/linux/man-pages/man7/udp.7.html says:
>  By default, Linux UDP does path MTU (Maximum Transmission Unit) discovery.  This means the kernel will keep track of the MTU to a specific target IP address and return EMSGSIZE when a UDP packet write exceeds it.  When this happens, the application should decrease the packet size.

Reply:
My understanding was `EMSGSIZE` is returned when we write a datagram larger than the MTU and the DF bit is set. So, `isSendMsgSizeErr` should return true when we receive `unix.EMSGSIZE` just like it does in sys_conn_df_linux.go

On the receive side, we never get any indication on macOS and Linux and the received datagram is just truncated.

Reply:
@sukunrt Sorry, I misunderstood what you were saying. I added the check for `isSendMsgSizeErr` on darwin.